lib/pos_transaction_register.go

```diff
@@ -0,0 +1,426 @@
+package lib
+
+import (
+	"fmt"
+	"github.com/emirpasic/gods/sets/treeset"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"math"
+	"math/big"
+)
+
+// ========================
+//	TransactionRegister
+// ========================
+
+// TransactionRegister is the primary transaction store of the PoS Mempool. The register is responsible for determining
+// the Fee-Time ordering of transactions. The operations supported by the register are: adding a transaction, removing
+// a transaction, iterating through all transactions in fee-time order, and retrieving all transactions ordered in
+// fee-time. The TransactionRegister doesn't perform any validation on the transactions, it just accepts the provided
+// MempoolTx and adds it to the appropriate FeeTimeBucket.
+type TransactionRegister struct {
+	// feeTimeBucketSet is a set of FeeTimeBucket objects. The set is ordered by the FeeTimeBucket's ranges, based on feeTimeBucketComparator.
+	feeTimeBucketSet *treeset.Set
+	// feeTimeBucketsByMinFeeMap is a map of FeeTimeBucket minimum fees to FeeTimeBucket objects. It is used to quickly find
+	// a FeeTimeBucket given its min fee.
+	feeTimeBucketsByMinFeeMap map[uint64]*FeeTimeBucket
+	// txnMembership is a set of transaction hashes. It is used to determine existence of a transaction in the register.
+	txnMembership *Set[BlockHash]
+	// totalTxnSize is the total size of all transactions in the register.
+	totalTxnSize uint64
+
+	params *DeSoParams
```

Comment:
Similar comment as with the GlobalParams. It's cleaner to pull the max mempool size params in the `NewTransactionRegister` constructor and store it locally.
```suggestion
	maxTxnSizeBytes uint64
```